docs/getting-started/ui/index.rst

```diff
@@ -0,0 +1,201 @@
+Authorization-Dependent UI Element
+==================================
+
+Using oso to control UI components
+----------------------------------
+
+The access control rules that you use to authorize data access in your
+application's backend often have implications for the frontend as well.
+For example, you may want to hide a button that links to another page if
+the current user doesn't have access to that page. Or perhaps you'd like
+to display an "edit" button only if the user is actually allowed to edit
+the resource in question.
+
+These are examples of what we at oso call "Authorization-Dependent UI
+Elements." In this guide we'll explain how you can use oso to implement
+these kinds of features in your app.
+
+    Note: we don't currently provide a version of oso that runs in the
+    browser. This guide covers how to query for information in the
+    backend that can be sent to your frontend service.
+
+Getting a user's allowed actions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you're familiar with oso, you know that oso policies contain
+``allow`` rules that specify that an **actor** is allowed to take an
+**action** on a **resource.**
+
+In many cases, like when you are authorizing a specific request in the
+backend, you'll just want to know if a specific
+``(actor, action, resource)`` combination is allowed, e.g., is
+``alice@gmail.com`` allowed to ``"READ"`` ``Expense{id: 1}``? For these
+yes/no queries, we provide the ``Oso.is_allowed()`` method.
+
+But when you're deciding what to display to Alice in the frontend, you
+may want more information. It can be especially useful to know **all the
+actions** that Alice is allowed to take on the ``Expense{id: 1}``
+resource, not just whether or not she can read it. In this case, you can
+use the ``Oso.get_allowed_actions()`` method to get a list of Alice's
+allowed actions and return them to your frontend.
+
+Let's look at an example. Imagine we have a GitHub-like app, that gives
+users access to repositories. The page to view a specific repository
+looks like this:
+
+.. image:: /getting-started/ui/a.png
+
+On this page there are several components that we may want to control
+based on the actions the current user is allowed to take. For example,
+we may want to hide the "Manage Access" or "Delete Repository" links
+depending on whether the user is allowed to take those actions.
+
+In this situation, the ``Oso.get_allowed_actions()`` method can be very
+helpful. The method returns a list of actions that a user is allowed to
+take on a specific resource. In our example, we call
+``get_allowed_actions()`` in the route handler for the "Show repository"
```

Comment:
Was leaving off the `Oso.` prefix intentional?

Reply:
In the code below it's not called on a top level `Oso` object.

Reply:
Doesn't `current_app.base_oso` return an `Oso` instance?

Reply:
yeah, I'm just saying were not calling it literally as `Oso.get_allowed_actions()`

Reply:
but yeah maybe it's still better to say that here for consistency

Reply:
Ah okay, I see your point. I read `Oso.get_allowed_actions()` as "a `get_allowed_actions()` method on an `Oso` object.